test/jdk/javax/swing/JFileChooser/4150029/bug4150029.java

```diff
@@ -21,73 +21,129 @@
-            try {
-                UIManager.setLookAndFeel("javax.swing.plaf.metal.MetalLookAndFeel");
-            } catch (Exception e) {
-                throw new RuntimeException(e);
+        try {
+            if (Platform.isOSX()) {
+                try {
+                    UIManager.setLookAndFeel("javax.swing.plaf.metal.MetalLookAndFeel");
+                } catch (Exception e) {
+                    throw new RuntimeException(e);
+                }
             }
-        }
 
-        String tmpDir = System.getProperty("java.io.tmpdir");
+            String tmpDir = System.getProperty("java.io.tmpdir");
 
-        if (tmpDir.length() == 0) {//'java.io.tmpdir' isn't guaranteed to be defined
-            tmpDir = System.getProperty("user.home");
+            //'java.io.tmpdir' isn't guaranteed to be defined
+            if (tmpDir.length() == 0) {
+                tmpDir = System.getProperty("user.home");
+            }
+            System.out.println("Temp directory: " + tmpDir);
+
+            testDir = new File(tmpDir, "testDir");
+            testDir.mkdir();
+            testDir.deleteOnExit();
+            System.out.println("Created directory: " + testDir);
+
+            subDir = new File(testDir, "subDir");
+            subDir.mkdir();
+            subDir.deleteOnExit();
+            System.out.println("Created sub-directory: " + subDir);
+
+            fileChooser = new JFileChooser(subDir);
+
+//            fileChooser.setFileSelectionMode(JFileChooser.DIRECTORIES_ONLY);
+
+            SwingUtilities.invokeAndWait(() -> {
+                frame = new JFrame("Backspace Shortcut for Directory Navigation Test");
+                frame.getContentPane().setLayout(new BorderLayout());
+                fileChooser = new JFileChooser(subDir);
+                fileChooser.setControlButtonsAreShown(false);
+                frame.getContentPane().add(fileChooser, BorderLayout.CENTER);
+                frame.pack();
+                frame.setLocationRelativeTo(null);
+                frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+                frame.setVisible(true);
+            });
+
+            doTesting();
+        } finally {
+            SwingUtilities.invokeAndWait(() -> {
+                if (frame != null) {
+                    frame.dispose();
+                }
+            });
         }
+    }
 
-        System.out.println("Temp directory: " + tmpDir);
-
-        File testDir = new File(tmpDir, "testDir");
-
-        testDir.mkdir();
-
-        File subDir = new File(testDir, "subDir");
-
-        subDir.mkdir();
-
-        System.out.println("Created directory: " + testDir);
-        System.out.println("Created sub-directory: " + subDir);
-
-        JFileChooser fileChooser = new JFileChooser(testDir);
+    private static void doTesting() {
+        Point p = frame.getLocationOnScreen();
+        robot.mouseMove(p.x + 200, p.y + 200);
+        robot.mousePress(InputEvent.BUTTON1_MASK);
```

Comment:
```suggestion
        robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
```

should we add `robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);` too ?